### PR TITLE
Enable all notification tests on Lomiri

### PR DESCRIPTION
**Pull request by tari01**
- tests/CMakeLists.txt: Add libayatana-common to notifications-test link libraries
- tests/notifications-test.cc: Enable all tests if run on Lomiri

### CODE REVIEW

File: tests/notifications-test.cc [hunk 1]
```diff
@@ -282,7 +285,13 @@ TEST_F(NotificationsTest, VolumeChanges) {
...
 }
 
-TEST_F(NotificationsTest, DISABLED_StreamChanges) {
+TEST_F(NotificationsTest, StreamChanges) {
+
+    if (!ayatana_common_utils_is_lomiri())
+    {
+        return;
+    }
+
```
Comment (sunweaver) [inline tests/notifications-test.cc:294]: Hmmm... this lets the test become a no-op test.

File: tests/notifications-test.cc [hunk 2]
```diff
@@ -323,7 +332,13 @@ TEST_F(NotificationsTest, DISABLED_StreamChanges) {
...
 }
 
-TEST_F(NotificationsTest, DISABLED_IconTesting) {
+TEST_F(NotificationsTest, IconTesting) {
+
+    if (!ayatana_common_utils_is_lomiri())
+    {
+        return;
+    }
+
```
Comment (sunweaver) [inline tests/notifications-test.cc:341]: dito

File: tests/notifications-test.cc [hunk 3]
```diff
@@ -360,7 +375,13 @@ TEST_F(NotificationsTest, DISABLED_IconTesting) {
...
 }
 
-TEST_F(NotificationsTest, DISABLED_ServerRestart) {
+TEST_F(NotificationsTest, ServerRestart) {
+
+    if (!ayatana_common_utils_is_lomiri())
+    {
+        return;
+    }
+
```
Comment (sunweaver) [inline tests/notifications-test.cc:384]: dito

File: tests/notifications-test.cc [hunk 4]
```diff
@@ -410,7 +431,13 @@ TEST_F(NotificationsTest, DISABLED_ServerRestart) {
...
 }
 
-TEST_F(NotificationsTest, DISABLED_HighVolume) {
+TEST_F(NotificationsTest, HighVolume) {
+
+    if (!ayatana_common_utils_is_lomiri())
+    {
+        return;
+    }
+
```
Comment (sunweaver) [inline tests/notifications-test.cc:440]: dito

File: tests/notifications-test.cc [hunk 5]
```diff
@@ -455,7 +482,13 @@ TEST_F(NotificationsTest, DISABLED_HighVolume) {
...
 }
 
-TEST_F(NotificationsTest, DISABLED_MenuHide) {
+TEST_F(NotificationsTest, MenuHide) {
+
+    if (!ayatana_common_utils_is_lomiri())
+    {
+        return;
+    }
+
```
Comment (sunweaver) [inline tests/notifications-test.cc:491]: dito

File: tests/notifications-test.cc [hunk 6]
```diff
@@ -488,7 +521,13 @@ TEST_F(NotificationsTest, DISABLED_MenuHide) {
...
 }
 
-TEST_F(NotificationsTest, DISABLED_ExtendendVolumeNotification) {
+TEST_F(NotificationsTest, ExtendendVolumeNotification) {
+
+    if (!ayatana_common_utils_is_lomiri())
+    {
+        return;
+    }
+
```
Comment (sunweaver) [inline tests/notifications-test.cc:530]: dito

File: tests/notifications-test.cc [hunk 7]
```diff
@@ -533,7 +572,12 @@ TEST_F(NotificationsTest, DISABLED_ExtendendVolumeNotification) {
     EXPECT_GVARIANT_EQ("@i 100", notev[0].hints["value"]);
 }
 
-TEST_F(NotificationsTest, DISABLED_TriggerWarning) {
+TEST_F(NotificationsTest, TriggerWarning) {
+
+    if (!ayatana_common_utils_is_lomiri())
+    {
+        return;
+    }
```
Comment (sunweaver) [inline tests/notifications-test.cc:580]: dito

**Review by sunweaver — CHANGES_REQUESTED** [review]
Hmmm... with this, you are not unit-testing anything in Lomiri really. This adjustment  only makes sense if you run the test twice, once with XDG_CURRENT_DESKTOP set to Lomiri (or some such) and once without. May guess is that our tests run without XDG_CURRENT_DESKTOP being set to anything.

As the notification-test.cc files does not use the Lomiri API for notifications directly, but a mock provided via notification-mock.h instead, the proper solution would probably be to repair the mock part of this test.

With the patch provided by this PR, we allude to the tests working ok, but what the test does is simply returning in the second line of test code. Not optimal.

Please resolve or postpone.

**Comment by tari01** [comment]
> Hmmm... with this, you are not unit-testing anything in Lomiri really. This adjustment only makes sense if you run the test twice, once with XDG_CURRENT_DESKTOP set to Lomiri (or some such) and once without. May guess is that our tests run without XDG_CURRENT_DESKTOP being set to anything.
> 
> As the notification-test.cc files does not use the Lomiri API for notifications directly, but a mock provided via notification-mock.h instead, the proper solution would probably be to repair the mock part of this test.
> 
> With the patch provided by this PR, we allude to the tests working ok, but what the test does is simply returning in the second line of test code. Not optimal.
> 
> Please resolve or postpone.

I discovered that these can be run on a Lomiri live system (SEGFAULTs everywhere else), hence the PR. It basically does what it's supposed to do - ignores the tests unless you are on a real Lomiri.

Since this is a hack, I'm OK if we simply close it. Doing it.

**Pull request closed without merging**